spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/security/saml2/Saml2RelyingPartyProperties.java

```diff
@@ -140,6 +140,11 @@ public void setCertificateLocation(Resource certificate) {
 		 */
 		private String entityId;
 
+		/**
+		 * Endpoint for discovery-based configuration.
+		 */
+		private String metadataUrl;
```

Comment:
Would that make sense to move that to a `metadata` sub-namespace the way we do for `singlesignon` and `verification`. Not sure if this could be extended in the future though.

Reply:
I don't know of a use case where there would be more to specify or configure regarding the identity provider's metadata. I'd recommend it stay at this level. In most cases, an application would either specify `metadata-url` or the rest of the properties.

That said, it occurs to me that there's metadata-based configuration support for OAuth 2.0 where the property is named `issuer-uri`. It may be better to call this `metadata-uri`.